providers/snowflake/database_grant.go

```diff
@@ -92,6 +89,9 @@ func (g *DatabaseGrantGenerator) InitResources() error {
 		}
 		allGrants = append(allGrants, grants...)
 	}
-	g.Resources = g.createResources(allGrants)
+	g.Resources, err = g.createResources(allGrants)
+	if err != nil {
```

Comment:
you can just `return err` since it is equivalent to checking if it is nil and then returning nil